src/main.rs

```diff
@@ -71,17 +98,49 @@ fn main() {
     let global = global_root.handle();
     let _ac = JSAutoCompartment::new(cx, global.get());
 
+    let rt_opts = unsafe { &mut *RuntimeOptionsRef(rt) };
+    rt_opts.set_baseline_(!js_options.disable_baseline);
+    rt_opts.set_ion_(!js_options.disable_ion);
+    rt_opts.set_asmJS_(!js_options.disable_asmjs);
+    rt_opts.set_extraWarnings_(js_options.enable_strict);
+    rt_opts.set_nativeRegExp_(!js_options.disable_native_regexp);
+    unsafe { JS_SetParallelParsingEnabled(rt, !js_options.disable_parallel_parsing); }
+    unsafe { JS_SetOffthreadIonCompilationEnabled(rt, !js_options.disable_offthread_compilation); }
+    unsafe { JS_SetGlobalJitCompilerOption(rt, JSJitCompilerOption::JSJITCOMPILER_BASELINE_WARMUP_TRIGGER,
+                                           if js_options.enable_baseline_unsafe_eager_compilation { 0i32 } else { -1i32 } as u32); }
+    unsafe { JS_SetGlobalJitCompilerOption(rt, JSJitCompilerOption::JSJITCOMPILER_ION_WARMUP_TRIGGER,
+                                           if js_options.enable_ion_unsafe_eager_compilation { 0i32 } else { -1i32 } as u32); }
+    rt_opts.set_werror_(js_options.enable_werror);
+    let mode = if !js_options.disable_incremental {
+        println!("incremental");
+        JSGCMode::JSGC_MODE_INCREMENTAL
+    } else if js_options.disable_gc_per_compartment {
+        println!("compartment");
+        JSGCMode::JSGC_MODE_COMPARTMENT
+    } else {
+        println!("global");
+        JSGCMode::JSGC_MODE_GLOBAL
+    };
+    unsafe { JS_SetGCParameter(rt, JSGCParamKey::JSGC_MODE, mode as u32); }
+    unsafe { JS_SetGCParameter(rt, JSGCParamKey::JSGC_COMPACTING_ENABLED, !js_options.disable_compacting as u32); }
+    unsafe { JS_SetGCParameter(rt, JSGCParamKey::JSGC_DYNAMIC_MARK_SLICE, !js_options.disable_dynamic_mark_slice as u32); }
+    unsafe { JS_SetGCParameter(rt, JSGCParamKey::JSGC_DYNAMIC_HEAP_GROWTH, !js_options.disable_dynamic_heap_growth as u32); }
```

Comment:
Here and above, the code gets quite wide, but again, no need to block the landing over this.